### ISSUE TITLE
你好，请教一下知识蒸馏方面的问题

### ISSUE THREAD

**Issue by xiexu666** [wangguanan/light-reid#24]
在代码中看到蒸馏部分的loss主要是
class KLLoss:
    '''KL Divergence'''

    def __init__(self, t=4):
        self.t = t

    def __call__(self, logits_s, logits_t):
        p_s = F.log_softmax(logits_s / self.t, dim=1)
        p_t = F.softmax(logits_t / self.t, dim=1)
        loss = F.kl_div(p_s, p_t, reduction='sum') * (self.t ** 2) / logits_s.shape[0]
        return loss

请问关于这里的原理相应论文可以用来学习吗，感谢~

**Comment by wangguanan** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    please refer https://arxiv.org/abs/1503.02531
    
        
            
        
    
    
          
        
            
    
        font{
            line-height: 1.6;
        }
    

    
        font{
            line-height: 1.6;
        }
    
Guan'an WangInstitute of Automation, Chinese Academy of Sciences,No. 95 Zhongguancun East Rd., Beijing, China.


        
    
     
    

    在2021年01月25日 22:06，xiexu666<notifications@github.com> 写道： 




在代码中看到蒸馏部分的loss主要是

class KLLoss:

'''KL Divergence'''

def __init__(self, t=4):

    self.t = t



def __call__(self, logits_s, logits_t):

    p_s = F.log_softmax(logits_s / self.t, dim=1)

    p_t = F.softmax(logits_t / self.t, dim=1)

    loss = F.kl_div(p_s, p_t, reduction='sum') * (self.t ** 2) / logits_s.shape[0]

    return loss



请问关于这里的原理相应论文可以用来学习吗，感谢~

**Comment by xiexu666** [comment]
> 
> font{ line-height: 1.6; } ul,ol{ padding-left: 20px; list-style-position: inside; } please refer https://arxiv.org/abs/1503.02531 font{ line-height: 1.6; } font{ line-height: 1.6; } Guan'an WangInstitute of Automation, Chinese Academy of Sciences,No. 95 Zhongguancun East Rd., Beijing, China. 在2021年01月25日 22:06，xiexu666<notifications@github.com> 写道： 在代码中看到蒸馏部分的loss主要是 class KLLoss: '''KL Divergence''' def __init__(self, t=4): self.t = t def __call__(self, logits_s, logits_t): p_s = F.log_softmax(logits_s / self.t, dim=1) p_t = F.softmax(logits_t / self.t, dim=1) loss = F.kl_div(p_s, p_t, reduction='sum') * (self.t ** 2) / logits_s.shape[0] return loss 请问关于这里的原理相应论文可以用来学习吗，感谢~ —You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

收到，非常感谢您的回复。
